### DISCUSSION TITLE
v0.16.0

### DISCUSSION THREAD

**Discussion by ledermann** [solectrus/solectrus#3405] [Updates]
## Major Changes

- Added PowerSplitter visualization (#2436)

## Minor Changes

- Settings: Added checkbox for "opportunity costs"
- Update dependencies (Rails 7.2.1 and more)

## Fixes

- Fixed handling prices of €0 (#3390)
- Fixed precision for smaller time ranges (#3399)
- Fixed caching for `all` (#3399)

**Full Changelog**: https://github.com/solectrus/solectrus/compare/v0.15.1...v0.16.0

<hr /><em>This discussion was created from the release <a href='https://github.com/solectrus/solectrus/releases/tag/v0.16.0'>v0.16.0</a>.</em>

**Comment by gereons** [comment]
Um das hier mal öffentlich zu sagen: mir gefällt die neue Freemium-Variante sehr gut, und ich hoffe sehr dass das für dich ein tragfähiges Modell für die Weiterentwicklung wird. Auch wenn ich selber nicht hätte abonnieren müssen (danke dafür!) hab ich's sofort und gerne getan.

**Comment by Zeppelin-PV** [comment]
... mir ging es genau so, - sofort, und sehr gerne. Ich weiß die Arbeit von Dir, Georg, zu schätzen.
Auch als "ahnungsloser" Standardnutzer, der zumindest Copy & Paste beherrscht, fühle ich mich wohl, in der Solectrus-Gemeinde.
Mach weiter so, Georg, und an alle die ihm mit tollen Ideen und Lösungen zur Seite stehen.

**Comment by muahsystems** [comment]
Ich habe mal eine Frage zur Umstellung auf Version v0.16.0
https://github.com/solectrus/solectrus/wiki/Umstellung-auf-neue-Sensor%E2%80%90Konfiguration
Hier vermisse ich tatsächlich eine gewisse Anleitung ... ich habe in der .env im Verzeichnis /solectrus geschaut, bei mir werden gar keine Sensoren definiert. Meine .env Datei ist aus dem Januar 2023.

**Reply by ledermann** [reply]
Die Sensor-Konfiguration wurde Juli 2024 mit `v0.15` eingeführt, mit `v0.16` hat sich nichts geändert. Über den von dir genannten Link gibt es Erklärungen dazu. Ich werde diese noch weiter ausbauen.

Der Dreh- und Angelpunkt ist das Docker-Log. Starte die Container neu (`docker compose down && docker compose up -d`) und gucke dann **genau** in die Logs, die beim Start geschrieben werden. Da finden sich dann Hinweise, was genau man tun muss, um auf die Sensoren-Konfiguration umzustellen. Es geht um Einträge in der `.env` und `docker-compose.yml`, die man ergänzen muss. Bei konkreten Fragen erstelle gerne ein neues Diskussionsthema, wir klären das dann.

**Comment by frizzle0815** [comment]
Hallo Georg,
auch ich habe gerade mein Abo abgeschlossen und würde mich freuen, wenn sich daraus irgendwann ein tragfähiges Modell für dich entwickelt!

Eine Frage: Ich betreibe zwei Instanzen von Solectrus (für die PV meiner Eltern und für meine eigene), die beide unter einem Konto registriert sind. Der Sponsoring Status wird aber nur bei der ersten Instanz erkannt. Muss man für jede Instanz ein eigenes Abo abschließen? Falls ja, wüsste ich aktuell nicht wie..

**Reply by ledermann** [reply]
Danke schön! :-)

Beim Thema "Betrieb von mehreren Instanzen" ist meine Haltung folgendermaßen: Wenn es sich wirklich um unterschiedliche PV-Anlagen auf verschiedenen Häusern handelt, dann ist ein entsprechend höherer Betrag angemessen. Also: Eine PV-Anlage = 5€/Monat, zwei PV-Anlagen = 10€/Monat usw. Man kann beim Klicken eines Abos die Menge angeben, die Vorgabe ist `1`. Stripe rechnet den Preis aus. Es ist also immer **ein** Abo, nur der Preis ist variabel. Die Menge lässt sich auch im Nachhinein ändern (in der Registrierung bei "Mein Abo" auf "Bearbeiten" klicken, in Stripe dann auf "Plan ändern"), wenn eine Instanz hinzukommt oder wegfällt.

Wer die gleiche PV-Anlage mit mehreren SOLECTRUS-Instanzen betreibt (z.B. als Backup), zahlt natürlich nur den **einfachen** Betrag.

Nun ist es so, dass ich die Sicherstellung der korrekten Menge noch nicht implementiert habe. Wer es also darauf anlegt, kann momentan die PV-Anlagen auf seinen zehn Villen mit einem 5€-Abo betreiben. Nachteil ist nur: Negatives Karma - und vielleicht prüft die Registrierungsverwaltung das irgendwann doch einmal ;-)

Eine kleine technische Ungereimtheit gibt es momentan noch, wenn man mehrere Instanzen betreibt: Direkt nach Abschließen des Abos wird man zu der Instanz zurückgeführt, von der man die Registrierung aufgerufen hat. Dadurch verschwindet dort sofort das gelbe Banner. Weitere Instanzen erfahren davon aber erst mal nichts und zeigen bis zu 12 Stunden lang das Banner weiterhin an. Du kannst das beheben, indem du bei bestehendem Abo in der Registrierungsverwaltung die betroffene Instanz auswählstt und dann den Button "Zur Instanz" wählst.
